src/r_util.cpp

```diff
@@ -487,6 +488,22 @@ namespace rhost {
             return Rf_ScalarInteger(static_cast<int>(lock_state));
         }
 
+        extern "C" SEXP fetch_file(SEXP path) {
+            fs::path fpath;
+            r_top_level_exec([&]() {
+                fpath = fs::u8path(Rf_translateCharUTF8(STRING_ELT(path, 0)));
+            }, __FUNCTION__);
+
+            if (!fpath.empty()) {
+                blobs::blob file_data;
+                blobs::append_from_file(file_data, fpath);
+                host::send_notification("!FetchFile", file_data, fpath.filename().string());
```

Comment:
The reason why plot needs its own notification is because it's triggered indirectly by the user, not directly by us. In this case, you're invoking this function directly via eval, so you might as well get its return value directly, too, using `REvaluationKind.RawResult`. Unless I'm missing something?